proposals/TestProposals.sol

```diff
@@ -41,20 +41,34 @@ contract TestProposals is Test {
     bool public doTeardown;
     bool public doValidate;
 
-    address public deployer = address(0x00000108);
+    address public deployer;
 
     function setUp() public {
-        addresses = new Addresses();
-
-        // Load proposals
-        proposals.push(Proposal(address(new zip000()))); /// Genesis token proposal
-        proposals.push(Proposal(address(new zip001()))); /// Wearables, Core, ADMIN_MULTISIG proposal
-        proposals.push(Proposal(address(new zip002()))); /// Timelock proposal
-        proposals.push(Proposal(address(new zip003()))); /// CGv1 proposal
-        proposals.push(Proposal(address(new zip004()))); /// TokenIds, MaxSupply and Capsule settings proposal
-        proposals.push(Proposal(address(new zip005()))); /// MaxSupply settings proposal
-        proposals.push(Proposal(address(new zip006()))); /// MaxSupply settings proposal
-        proposals.push(Proposal(address(new zip007()))); /// MaxSupply settings proposal
+        string memory environment = vm.envOr("ENVIRONMENT", string("localnet"));
+        string memory addressPath = string(abi.encodePacked("proposals/Addresses/", environment, ".json"));
+        addresses = new Addresses(addressPath);
+
+        bool bootstrap = vm.envOr("BOOTSTRAP", false);
+
+        // Admin access for deployer(ZTX_DEPLOYER) is revoked on mainnet
+        if (block.chainid == 42161) {
+            deployer = addresses.getAddress("ADMIN_MULTISIG");
+        } else {
+            deployer = addresses.getAddress("DEPLOYER");
+        }
+
+        if (block.chainid == 31337 || bootstrap) {
```

Comment:
Why do we want this BOOTSTRAP environment variable?

Reply:
I added this in case if we have to bootstrap a new testing environment on testnet, in this case all the zips must be simulated. If we will run the following command a new system will be bootstrapped.
`BOOTSTRAP=true forge test --fork-url $RPC_URL --match-contract TestProposals -vvv --keystore <Path> --password <Password>`
where keystore has deployer private key.

We can make a separate script for this as well. Should I add a script also?

Reply:
Ok, but this is still incomplete because you have to go and adjust the `DeployProposal.s.sol` for running each proposal.

Reply:
updated here [814fc87](https://github.com/ZTX-Foundation/tuxedo/pull/203/commits/814fc87a22c8a93aead5e38c4d33d020fc42ca73)